File: Server/apiplate/Domain/Models/Offer.cs
using System;
using System.ComponentModel.DataAnnotations.Schema;

namespace apiplate.Domain.Models
{
public class Offer : BaseModel
{
    public string Title { get; set; }
    public Image Image { get; set; }
    public int Discount { get; set; }
    public DateTime StartDate { get; set; }
    public DateTime EndDate { get; set; }
}
}

File: Client/babcare/lib/models/offer.dart
import 'image.dart';

class Offer {
  String? title;
  Image? image;
  int? categoryId;
  String? startDate;
  String? endDate;
  int? discount;
  int? id;

  Offer(
      {this.title,
      this.image,
      this.categoryId,
      this.startDate,
      this.endDate,
      this.discount,
      this.id});

  Offer.fromJson(Map<String, dynamic> json) {
    title = json['title'];
    image = json['image'] != null ? Image.fromJson(json['image']) : null;
    categoryId = json['categoryId'];
    startDate = json['startDate'];
    endDate = json['endDate'];
    discount = json['discount'];
    id = json['id'];
  }
}
